### ISSUE TITLE
Port twisted.internet.cfreactor to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8838]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8838|
|Type|enhancement|
|Created|2016-09-24 20:41:18Z|
|Branch|https://github.com/twisted/twisted/tree/8838-cfreactor-py3|

<details><summary>Searchable metadata</summary>

```
trac-id__8838 8838
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8838_cfreactor_py3 8838-cfreactor-py3
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1474749678198879 1474749678198879
changetime__1476096435172703 1476096435172703
version__None None
owner__Amber_Brown__hawkowl_____ Amber Brown <hawkowl@...>

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

builds are green on the pr, pls review

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_3"></a>|@glyph set owner to @hawkowl|
|-|-|

See github review; please address those comments and resubmit.

https://github.com/twisted/twisted/pull/551#pullrequestreview-1460169

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl removed owner|
|-|-|

I think I've resolved all the points for this. Please review.

**Comment by twisted-trac** [comment]
|[<img alt="markrwilliams's avatar" src="https://avatars.githubusercontent.com/u/704247?s=50" width="50" height="50">](https://github.com/markrwilliams)<a name="note_5"></a>|@markrwilliams commented|
|-|-|

Reviewed here: https://github.com/twisted/twisted/pull/551#pullrequestreview-3449638

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_6"></a>|@hawkowl set owner to @hawkowl<br>@hawkowl set status to `closed`|
|-|-|

In changeset 483913fbdfcd35b7a02a0050a72e79cbdbea6e36
```
#!CommitTicketReference repository="" revision="483913fbdfcd35b7a02a0050a72e79cbdbea6e36"
Merge 8838-cfreactor-py3: Port CFReactor to Python 3 & make it supported

Author: hawkowl
Reviewers: glyph, markrwilliams
Fixes: #8838
```
